### PR TITLE
show cached user info when network down

### PR DESCRIPTION
**Pull request by Logan676**
the app crashed when trying to open settings menu if network down. To resolve this error, cached user info and show cache when offline.

Also show a progress bar at the top right corner of SettingsActivity when loading.

### CODE REVIEW

**Comment by Logan676** [comment]
@lins05 I changed to cache account info by a hash map.

File: src/com/seafile/seadroid2/ui/fragment/SettingsPreferenceFragment.java [hunk 1]
```diff
@@ -42,12 +40,18 @@
...
 
 public class SettingsPreferenceFragment extends CustomPreferenceFragment implements
         OnPreferenceChangeListener, OnPreferenceClickListener {
     private static final String DEBUG_TAG = "SettingsPreferenceFragment";
 
     public static final String EXTRA_CAMERA_UPLOAD = "com.seafile.seadroid2.camera.upload";
+
+    // Account Info
+    public static final String SPACE_USAGE_SEPERATOR = " / ";
+    private Map<String, AccountInfo> accountInfoMap = Maps.newHashMap();
```
Comment (lins05) [inline src/com/seafile/seadroid2/ui/fragment/SettingsPreferenceFragment.java]: `accountInfoMap` should be a static member variable.

File: src/com/seafile/seadroid2/ui/fragment/SettingsPreferenceFragment.java [hunk 2]
```diff
@@ -426,23 +448,53 @@ protected AccountInfo doInBackground(Account... params) {
...
     }
 
+    public String getCurrentUserIdentifier() {
+        if (settingsMgr.getCurrentAccount() == null)
+            return "";
+
+        Account account = settingsMgr.getCurrentAccount();
+        String username = account.getEmail();
+        String server = Utils.stripSlashes(account.getServerHost());
+        return Utils.assembleUserName(username, server);
```
Comment (lins05) [inline src/com/seafile/seadroid2/ui/fragment/SettingsPreferenceFragment.java]: Why not use `Account.getSigature`?

File: src/com/seafile/seadroid2/ui/fragment/SettingsPreferenceFragment.java [hunk 3]
```diff
@@ -116,12 +125,20 @@ public void onViewCreated(View view, Bundle savedInstanceState) {
...
-        // AccountInfo actInfo = accountMgr.getCurrentAccountInfo();
         spaceAvailablePref = findPreference(SettingsManager.SETTINGS_ACCOUNT_SPACE_KEY);
+
+        String identifier = getCurrentUserIdentifier();
+        AccountInfo info = getAccountInfoByIdentifier(identifier);
+        if (info != null) {
+            String spaceUsed = getSpaceUsed(info.getUsage(), info.getTotal());
+            spaceAvailablePref.setSummary(spaceUsed);
+        } else
+            spaceAvailablePref.setSummary(getString(R.string.not_available));
```
Comment (lins05) [inline src/com/seafile/seadroid2/ui/fragment/SettingsPreferenceFragment.java]: "Not available" is more suitable when an error happens. What about "loading"?

File: src/com/seafile/seadroid2/ui/fragment/SettingsPreferenceFragment.java (hunk 2 above)
Comment (Logan676) [inline src/com/seafile/seadroid2/ui/fragment/SettingsPreferenceFragment.java]: Because the identifier will be displayed in subtitle of User Info Menu.

**Pull request merged**